File: ios/RBDarts/Features/Practice/PracticeDiagnostics.swift
import Foundation

public struct PracticeDiagnostics {
    private let reporter: DiagnosticsReporting

    public init(reporter: DiagnosticsReporting = PrivacySafeDiagnostics()) {
        self.reporter = reporter
    }

    public func recordInvalidPracticeScore(target: Int) {
        reporter.record(DiagnosticEvent(name: "practice_invalid_score", severity: .warning, attributes: ["target": "\(target)"]))
    }

    public func recordSaveFailure(target: Int) {
        reporter.record(DiagnosticEvent(name: "practice_save_failed", severity: .error, attributes: ["target": "\(target)"]))
    }
}

File: android/app/src/main/java/com/rbdarts/feature/practice/PracticeDiagnostics.kt
package com.rbdarts.feature.practice

import com.rbdarts.core.observability.DiagnosticEvent
import com.rbdarts.core.observability.DiagnosticSeverity
import com.rbdarts.core.observability.DiagnosticsReporter
import com.rbdarts.core.observability.PrivacySafeDiagnostics

class PracticeDiagnostics(private val reporter: DiagnosticsReporter = PrivacySafeDiagnostics()) {
    fun recordInvalidPracticeScore(target: Int) {
        reporter.record(DiagnosticEvent("practice_invalid_score", DiagnosticSeverity.Warning, mapOf("target" to "$target")))
    }

    fun recordSaveFailure(target: Int) {
        reporter.record(DiagnosticEvent("practice_save_failed", DiagnosticSeverity.Error, mapOf("target" to "$target")))
    }
}
